src/NHSD.BuyingCatalogue.Identity.Api/Services/PasswordService.cs

```diff
@@ -0,0 +1,93 @@
+        public PasswordService(
+            IEmailService emailService,
+            PasswordResetSettings settings,
+            UserManager<ApplicationUser> userManager)
+        {
+            _emailService = emailService ?? throw new ArgumentNullException(nameof(emailService));
+            _settings = settings ?? throw new ArgumentNullException(nameof(settings));
+            _userManager = userManager ?? throw new ArgumentNullException(nameof(userManager));
+        }
+
+        /// <summary>
+        /// Generates a password reset token for the user with the
+        /// provided <paramref name="emailAddress"/>.
+        /// </summary>
+        /// <param name="emailAddress">The e-mail of the user to generate the password
+        /// reset token for.</param>
+        /// <returns>A <see cref="PasswordResetToken"/> if the user was found;
+        /// otherwise, <see lagref="null"/>.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="emailAddress"/> is <see langref="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="emailAddress"/> is empty or white space.</exception>
+        public async Task<PasswordResetToken> GeneratePasswordResetTokenAsync(string emailAddress)
+        {
+            if (emailAddress is null)
+                throw new ArgumentNullException(nameof(emailAddress));
+
+            if (string.IsNullOrWhiteSpace(emailAddress))
+                throw new ArgumentException($"{emailAddress} must be provided", nameof(emailAddress));
+
+            var user = await _userManager.FindByEmailAsync(emailAddress);
+
+            return user == null
+                ? null
+                : new PasswordResetToken(await _userManager.GeneratePasswordResetTokenAsync(user), user);
+        }
+
+        /// <summary>
+        /// Sends a password reset e-mail to the specified <paramref name="user"/>.
+        /// </summary>
+        /// <param name="user">The user to send the e-mail to.</param>
+        /// <param name="callback">The callback URL to handle
+        /// the password reset.</param>
+        /// <returns>An asynchronous task context.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="user"/> is <see langref="null"/>.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="callback"/> is <see langref="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="callback"/> is empty or white space.</exception>
+        public async Task SendResetEmailAsync(ApplicationUser user, string callback)
+        {
+            if (user is null)
+                throw new ArgumentNullException(nameof(user));
+
+            if (callback is null)
+                throw new ArgumentNullException(nameof(callback));
+
+            if (string.IsNullOrWhiteSpace(callback))
+                throw new ArgumentException($"{callback} must be provided", nameof(callback));
```

Comment:
this is going to say " should be provided" as callback will be null when it creates that string.

Reply:
Should be `nameof(callback)` – I'll fix.